homeassistant/components/sonarr/coordinator.py

```diff
@@ -0,0 +1,148 @@
+"""Data update coordinator for the Sonarr integration."""
+from __future__ import annotations
+
+from abc import abstractmethod
+from datetime import timedelta
+from typing import TypeVar, Union, cast
+
+from aiopyarr import (
+    Command,
+    Diskspace,
+    SonarrCalendar,
+    SonarrQueue,
+    SonarrSeries,
+    SonarrWantedMissing,
+    exceptions,
+)
+from aiopyarr.models.host_configuration import PyArrHostConfiguration
+from aiopyarr.sonarr_client import SonarrClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+import homeassistant.util.dt as dt_util
+
+from .const import CONF_UPCOMING_DAYS, CONF_WANTED_MAX_ITEMS, DOMAIN, LOGGER
+
+T = TypeVar(
+    "T",
+    bound=Union[
+        list[SonarrCalendar],
+        list[Command],
+        list[Diskspace],
+        SonarrQueue,
+        list[SonarrSeries],
+        str,
+        SonarrWantedMissing,
+    ],
+)
```

Comment:
`T` isn't a good name for this TypeVar. Especially since it will be used in other modules too. Maybe something like `SonarrDataT`?